src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php

```diff
@@ -33,20 +32,7 @@ public function isFresh($timestamp)
             return false;
         }
 
-        // has the file *not* been modified? Definitely fresh
-        if (@filemtime($this->filePath) <= $timestamp) {
-            return true;
-        }
-
-        try {
-            $reflectionClass = new \ReflectionClass($this->class);
-        } catch (\ReflectionException $e) {
-            // the class does not exist anymore!
-
-            return false;
-        }
-
-        return AutowirePass::createResourceForClass($reflectionClass);
+        return @filemtime($this->filePath) <= $timestamp;
```

Comment:
@HeahDude thanks for fixing the test failure on #18144 - I didn't realize I had left an error!

But, this is wrong! This simplifies this resource down to be basically identical to the `SelfCheckingResourceInterface`.
The original last line *was* a mistake - I added a bug in my last commit - this was the mistake: https://github.com/symfony/symfony/pull/18144/commits/49271e86f999ad84bb4f8787b844973f7c22f4fc#diff-04e2904117929f6ffe784d5be3cb1103R49

So, this should be reverted, but and the last line corrected to be:

```php
return AutowirePass::createResourceForClass($reflectionClass); == $this;
```

Could you make a PR for that?